src/aiida/cmdline/commands/cmd_group.py

```diff
@@ -145,41 +145,161 @@ def group_move_nodes(source_group, target_group, force, nodes, all_entries):
+    contains,
+    node,
+    **traversal_rules,
+):
+    """Delete groups and (optionally) the nodes it contains."""
+    from tabulate import tabulate
+
     from aiida import orm
     from aiida.tools import delete_group_nodes
 
-    if not (force or dry_run):
-        click.confirm(f'Are you sure you want to delete {group}?', abort=True)
-    elif dry_run:
-        echo.echo_report(f'Would have deleted {group}.')
+    if groups and (
+        all_users or user or past_days or startswith or endswith or contains or node or type_string != 'core'
+    ):
+        echo.echo_critical('Cannot specify both GROUPS and any of the other filters.')
+        return
+
+    if not groups:
+        import datetime
+
+        from aiida.common import timezone
+        from aiida.common.escaping import escape_for_sql_like
+
+        builder = orm.QueryBuilder()
+        filters = {}
+
+        if '%' in type_string or '_' in type_string:
+            filters['type_string'] = {'like': type_string}
+        else:
+            filters['type_string'] = type_string
+
+        # Creation time
+        if past_days:
+            filters['time'] = {'>': timezone.now() - datetime.timedelta(days=past_days)}
+
+        # Query for specific group labels
+        filters['or'] = []
+        if startswith:
+            filters['or'].append({'label': {'like': f'{escape_for_sql_like(startswith)}%'}})
+        if endswith:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(endswith)}'}})
+        if contains:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(contains)}%'}})
 
-    if delete_nodes:
+        builder.append(orm.Group, filters=filters, tag='group', project='*')
+
+        # Query groups that belong to specific user
+        if user:
+            user_email = user.email
+        else:
+            # By default: only groups of this user
+            user_email = orm.User.collection.get_default().email
 
-        def _dry_run_callback(pks):
-            if not pks or force:
-                return False
-            echo.echo_warning(f'YOU ARE ABOUT TO DELETE {len(pks)} NODES! THIS CANNOT BE UNDONE!')
-            return not click.confirm('Do you want to continue?', abort=True)
+        # Query groups that belong to all users
+        if not all_users:
+            builder.append(orm.User, filters={'email': user_email}, with_group='group')
 
-        _, nodes_deleted = delete_group_nodes([group.pk], dry_run=dry_run or _dry_run_callback, **traversal_rules)
-        if not nodes_deleted:
-            # don't delete the group if the nodes were not deleted
+        # Query groups that contain a particular node
+        if node:
+            builder.append(orm.Node, filters={'id': node.pk}, with_group='group')
+
+        groups = builder.all(flat=True)
+        if not groups:
+            echo.echo_report('No groups found matching the specified criteria.')
             return
 
-    if not dry_run:
+    if groups:
```

Comment:
I don't think this conditional is necessary? If `groups` is falsey, the previous conditional will already have called `return` so this block would never be hit

Reply:
Good that you mentioned, I realized this was buggy.
I wanted to catch the case with no input `verdi group delete`, but obviously was not working. 
Now it's ok, I added respective test as well.